WooCommerce/Classes/ViewRelated/Dashboard/Settings/Settings/SettingsViewController.swift

```diff
@@ -59,6 +59,12 @@ final class SettingsViewController: UIViewController {
         viewModel.onViewDidLoad()
     }
 
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        viewModel.reloadSettings()
```

Comment:
I'm wondering if we want to reload all settings sections every time we land on the settings screen? Just raising the question because you're much experienced:

If `viewWillAppear` is called each time we land on the page, and we're reloading all `Sections` on the main thread, I was wondering if there's something in these `Sections` that is happening in a background thread and could cause data to not be updated properly, for example `isJetpackCPConnected` seems to be coming from Networking.

Reply:
Good point here @iamgabrielma :) 

> If viewWillAppear is called each time we land on the page, and we're reloading all Sections on the main thread, I was wondering if there's something in these Sections that is happening in a background thread and could cause data to not be updated properly, for example isJetpackCPConnected seems to be coming from Networking.

Actually, I do not see any call that might make us refresh in a background thread here :) `stores.sessionManager.defaultSite?.isJetpackCPConnected == true` is not changing the thread, it is just checking for that property in the same thread as it was called, presumably in the main one. As the method is totally synchronous, we can be certain about this assumption.
Nevertheless, I considered that it might be a little bit overkill to load all settings when we just want to hide the banner. After that consideration, I assumed that the simplicity of code and easiness of effort is a good trade-off for this. In any case, we are not applying large computations for reloading the settings, so after all it might not be that expensive.